scripts/prepdocs.py

```diff
@@ -41,24 +45,42 @@
 search_creds = default_creds if args.searchkey == None else AzureKeyCredential(args.searchkey)
 if not args.skipblobs:
     storage_creds = default_creds if args.storagekey == None else args.storagekey
+if not args.localpdfparser:
+    # check if Azure Form Recognizer credentials are provided
+    if args.formrecognizerservice == None:
+        print("Error: Azure Form Recognizer service is not provided. Please provide formrecognizerservice or use --localpdfparser for local pypdf parser.")
+        exit(1)
+    formrecognizer_creds = default_creds if args.formrecognizerkey == None else AzureKeyCredential(args.formrecognizerkey)
 
-def blob_name_from_file_page(filename, page):
-    return os.path.splitext(os.path.basename(filename))[0] + f"-{page}" + ".pdf"
+def blob_name_from_file_page(filename, page = 0):
+    if os.path.splitext(filename)[1].lower() == ".pdf":
+        return os.path.splitext(os.path.basename(filename))[0] + f"-{page}" + ".pdf"
+    else:
+        return os.path.basename(filename)
 
-def upload_blobs(pages):
+def upload_blobs(filename):
     blob_service = BlobServiceClient(account_url=f"https://{args.storageaccount}.blob.core.windows.net", credential=storage_creds)
     blob_container = blob_service.get_container_client(args.container)
     if not blob_container.exists():
         blob_container.create_container()
-    for i in range(len(pages)):
-        blob_name = blob_name_from_file_page(filename, i)
-        if args.verbose: print(f"\tUploading blob for page {i} -> {blob_name}")
-        f = io.BytesIO()
-        writer = PdfWriter()
-        writer.add_page(pages[i])
-        writer.write(f)
-        f.seek(0)
-        blob_container.upload_blob(blob_name, f, overwrite=True)
+
+    # if file is PDF split into pages and upload each page as a separate blob
+    if os.path.splitext(filename)[1].lower() == ".pdf":
+        reader = PdfReader(filename)
+        pages = reader.pages
+        for i in range(len(pages)):
+            blob_name = blob_name_from_file_page(filename, i)
+            if args.verbose: print(f"\tUploading blob for page {i} -> {blob_name}")
+            f = io.BytesIO()
+            writer = PdfWriter()
+            writer.add_page(pages[i])
+            writer.write(f)
+            f.seek(0)
+            blob_container.upload_blob(blob_name, f, overwrite=True)
+    else:
```

Comment:
We might want to not upload blobs in the non-PDF case (and maybe print a warning) if Form Recognizer is not enabled, since we won't be able to parse these.

Reply:
Yes, I've seen this issue but I've left it as is for now to make code simpler.